### ISSUE TITLE
Starfield Not Rendering At Longer Distances

### ISSUE THREAD

**Issue by MadDoctorTolliver** [TheCodex6824/ThaumicAugmentation#318]
Not sure if this is a bug with this mod or some other rendering issue with Minecraft in general, but Starfield Glass does not render the starfield at long distances. I was planning to use the Starfield Glass to make a large "room" appear extradimensional for decorative purposes, because it's perfect for that sort of thing, but if the player is too far from the glass, it simply shows up as clear glass. It's as if the starfield itself only renders out to a specific radius from the player.

I tested this on a world using the following Minecraft setup:

Minecraft: 1.12.2
Forge Mod Loader: 14.23.5.2860

Baubles: 1.12-1.5.2
Connected Textures Mod: CTM-MC1.12.2-1.0.2.31
Thaumcraft: Thaumcraft-1.12.2-6.1.BETA26
Thaumic Augmentation: ThaumicAugmentation-1.12.2-2.1.7

Attached are screenshots taken from within the "room" and a test save for your perusal.

[Testing.zip](https://github.com/TheCodex6824/ThaumicAugmentation/files/9691167/Testing.zip)

![2022-10-01_16 29 47](https://user-images.githubusercontent.com/114832658/193427447-588929bd-6053-4159-8764-11eadb312ca6.png)

![2022-10-01_16 30 01](https://user-images.githubusercontent.com/114832658/193427451-be7bc619-6fb9-4b4f-af2a-2d3345870268.png)

![2022-10-01_16 30 09](https://user-images.githubusercontent.com/114832658/193427453-60596d77-387c-4d11-b205-a72a5ba4f97b.png)

![2022-10-01_16 30 22](https://user-images.githubusercontent.com/114832658/193427454-726f9304-9434-41b4-b839-17948fa5a1b9.png)

**Comment by MadDoctorTolliver** [comment]
A slight edit: after a quick rebuild by encasing the same size (200x100x200) "room" with obsidian, an x-ray effect is observed along with the disappearance of the starfield. Given the outer layer of solid blocks, the sky should not be visible through the starfield glass, and yet, the sky is visible through the distant wall, as is shown below.

![2022-10-01_21 32 35](https://user-images.githubusercontent.com/114832658/193434132-d1481d39-4465-495d-be1d-dc858775dad7.png)

**Comment by TheCodex6824** [comment]
I think you set a new record for the largest starfield room - the current distance limit for rendering the starfield effect is 128 blocks, and nobody seems to have run into that limit before. I was afraid of making it larger in case it caused lag, but as long as the shader batching I tried to do is helping performance like I intended, it might not be too bad. How is your FPS in the room, and what is your system specs?

Either way, I think I'll have to go back and make all of these distances configurable, as long as I can also figure out what's going on with that xray thing.

**Comment by MadDoctorTolliver** [comment]
Hmmm. Lag was not something I had considered, admittedly. I did notice a fairly sizable FPS drop inside the room (60 outside, ~20 inside), but no real lag. The only real reason I noticed the FPS drop was because I had the F3 details open.

My specs are reasonable, but not outlandish (AMD Ryzen 7 3700X 8-Core Processor,                3.59 GHz; 32GB RAM; NVIDIA Geforce GTX 1660 Ti), but if it's causing that level of FPS drop on my system, then I probably need to rethink my build plans. I guess my idea of "train station floating in a starry void" will remain a pipe dream for now. Unfortunate, but such is life, I suppose.
